### PR TITLE
Add Qwen3-VL models to model zoo

### PR DESCRIPTION
**Pull request by CharlesCNorton**
## Summary

- Add Qwen3-VL models to the model zoo
- Create `fiftyone/utils/qwen3_vl.py` wrapper for 2D object detection

## Models Added

| Model | Size | Description |
|-------|------|-------------|
| `qwen3-vl-2b-instruct-torch` | 4.5 GB | Compact, ~6-8 GB VRAM |
| `qwen3-vl-4b-instruct-torch` | 9 GB | Balanced performance |
| `qwen3-vl-8b-instruct-torch` | 18 GB | High accuracy |

## Usage

```python
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart", max_samples=5)

model = foz.load_zoo_model("qwen3-vl-2b-instruct-torch")
dataset.apply_model(model, label_field="detections")

session = fo.launch_app(dataset)
```

Detect specific classes:

```python
model = foz.load_zoo_model(
    "qwen3-vl-2b-instruct-torch",
    classes=["person", "car", "dog"],
)
```

## Test plan

- [x] Model loading and inference on GPU (RTX 6000 Ada)
- [x] Output bounding boxes as `fol.Detections`
- [x] Coordinate conversion from 0-1000 scale to normalized [x, y, w, h]
- [x] Handle empty responses
- [x] Multiple input types (PIL, NumPy, Tensor)
- [x] `classes` parameter for targeted detection
- [x] Apply to FiftyOne quickstart dataset
- [x] Launch FiftyOne app with detection results
- [x] Parse JSON with/without markdown formatting

## Summary by CodeRabbit

* **New Features**
  * Added vision–language model integration for image inference: end-to-end object detection and optional image embeddings with configurable default model, prompt, and class labels.

* **Bug Fixes / Reliability**
  * More robust output parsing for JSON/markdown outputs; normalized, clamped, and validated bounding boxes; improved handling of malformed, out‑of‑range, inverted, and zero‑size boxes.

* **Tests**
  * Comprehensive unit tests covering parsing, edge cases, batching, prompts, configs, and embedding vs detection modes.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: fiftyone/utils/qwen3_vl.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+            bbox = obj.get("bbox_2d")
+
+            if bbox is None or len(bbox) != 4:
+                continue
+
+            x1, y1, x2, y2 = bbox
+            x1 = float(np.clip(x1 / 1000.0, 0.0, 1.0))
+            y1 = float(np.clip(y1 / 1000.0, 0.0, 1.0))
+            x2 = float(np.clip(x2 / 1000.0, 0.0, 1.0))
+            y2 = float(np.clip(y2 / 1000.0, 0.0, 1.0))
```
Comment (lorenzoriano) [inline fiftyone/utils/qwen3_vl.py:120]: I'm sure you know what you're doing here (and it's correct), but please add some comments to explain what is going on. Also up above with json parsing

File: fiftyone/utils/qwen3_vl.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            y1 = y1 / 1000.0
+            x2 = x2 / 1000.0
+            y2 = y2 / 1000.0
+
+            w = x2 - x1
+            h = y2 - y1
+
+            if w <= 0 or h <= 0:
+                continue
+
```
[the remark this answers is not in the document]
Comment (lorenzoriano) [reply fiftyone/utils/qwen3_vl.py:127]: @CharlesCNorton is this valid?

File: fiftyone/utils/qwen3_vl.py [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+        return np.vstack(embeddings)
+
+    def _prepare_image(self, img):
+        """Convert image to PIL format for processor."""
+        from PIL import Image as PILImage
+
+        if isinstance(img, torch.Tensor):
+            img = img.cpu().numpy()
+            if img.shape[0] in (1, 3, 4):
+                img = np.transpose(img, (1, 2, 0))
```
Comment (lorenzoriano) [inline fiftyone/utils/qwen3_vl.py:373]: Same concern, it looks very ad hoc and should be well documented. Also what if you have HWC images with height 1,3,4 ?

File: tests/unittests/utils/test_qwen3_vl.py [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+|
+"""
+
+import pytest
+import numpy as np
+
+import fiftyone.core.labels as fol
+
+
+class TestQwen3VLOutputProcessor:
```
Comment (lorenzoriano) [inline tests/unittests/utils/test_qwen3_vl.py:15]: Can you test for  HWC images with height 1,3,4 ? An HWC image of shape (3, 224, 3) would have img.shape[0] == 3, triggering the transpose and producing a malformed (224, 3, 3)

File: fiftyone/utils/qwen3_vl.py (hunk 1 above)
Comment (CharlesCNorton) [reply fiftyone/utils/qwen3_vl.py:120]: done!

File: fiftyone/utils/qwen3_vl.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (CharlesCNorton) [reply fiftyone/utils/qwen3_vl.py:127]: yes they output bbox_2d in 0-1000 scale.

File: tests/unittests/utils/test_qwen3_vl.py (hunk 4 above)
Comment (CharlesCNorton) [reply tests/unittests/utils/test_qwen3_vl.py:15]: added!

File: fiftyone/utils/qwen3_vl.py (hunk 3 above)
Comment (CharlesCNorton) [reply fiftyone/utils/qwen3_vl.py:373]: fixed and documented!
